src/components/calendar/MainCalendar.tsx

```diff
@@ -1,35 +1,36 @@
-import { useDisclosure } from '@chakra-ui/react'
+import { Button, useDisclosure } from '@chakra-ui/react'
+import styled from '@emotion/styled'
 import type { EventClickArg } from '@fullcalendar/core'
-import FullCalendar from '@fullcalendar/react' // must go before plugins
 import dayGridPlugin from '@fullcalendar/daygrid'
 import interactionPlugin from '@fullcalendar/interaction'
+import FullCalendar from '@fullcalendar/react'; // must go before plugins
 import timeGridPlugin from '@fullcalendar/timegrid'
-import styled from '@emotion/styled'
+import { useState } from 'react'
 import { ModalContext } from '~/context/ModalContext'
-import { Button } from '@chakra-ui/react'
 import { trpc } from '~/utils/trpc'
-import EventModal from './EventModal'
-import { useContext, useState } from 'react'
 import CreateEventModal from './CreateEventModal'
+import EventModal from './EventModal'
 
 export const StyleWrapper = styled.div``
 
 const MainCalendar = () => {
-  const eventModal = useContext(ModalContext)
   const { data } = trpc.event.populateCalendar.useQuery()
-  const { onOpen, isOpen, onClose } = useDisclosure()
+  const { onOpen: onOpenDisplayEvent, isOpen: isOpenDisplayEvent, onClose: onCloseDisplayEvent } = useDisclosure()
+  const { onOpen: onOpenCreateEvent, isOpen: isOpenCreateEvent, onClose: onCloseCreateEvent } = useDisclosure()
+
+  const [displayEventId, setDisplayEventId] = useState<string>("")
+  const [createEventId, setCreateEventId] = useState<string>("")
 
-  const [id, setId] = useState<string>("")
-  
 
   const handleEventClick = (info: EventClickArg) => {
-    setId(info.event.id)
-    onOpen()
+    setDisplayEventId(info.event.id)
+    onOpenDisplayEvent()
   }
 
   const handleCreateEvent = () => {
-    setId(eventModal.id)
-    onOpen()
+    const uniqueId = Date.now().toString();
```

Comment:
I'm not sure what this ID is supposed to be, but it triggers the following error
❌ tRPC failed on event.getEvent: TRPCError: Cannot read properties of undefined (reading 'filter')